Core/include/Acts/TrackFinding/MeasurementSelector.hpp

```diff
@@ -82,166 +82,28 @@ class MeasurementSelector {
-    }
-
-    std::sort(candidates.begin(), trackStateIterEnd,
-              [](const auto& tsa, const auto& tsb) {
-                return tsa.chi2() < tsb.chi2();
-              });
-
-    // use |eta| of best measurement to select numMeasurementsCut
-    const auto numMeasurementsCut = VariableCut<traj_t>(
-        *candidates.begin(), cuts, cuts->numMeasurementsCutOff, logger);
-
-    if (static_cast<std::size_t>(std::distance(
-            candidates.begin(), trackStateIterEnd)) > numMeasurementsCut &&
-        numMeasurementsCut > 0) {
-      trackStateIterEnd = std::next(candidates.begin(), numMeasurementsCut);
-    }
-
-    ACTS_VERBOSE("Number of selected measurements: "
-                 << std::distance(candidates.begin(), trackStateIterEnd)
-                 << ", max: " << numMeasurementsCut);
-
-    isOutlier = false;
-    return std::pair{candidates.begin(), trackStateIterEnd};
-  }
+         bool& isOutlier, const Logger& logger) const;
 
  private:
   template <typename traj_t, typename cut_value_t>
   static cut_value_t VariableCut(
       const typename traj_t::TrackStateProxy& trackState,
       const Acts::MeasurementSelector::Config::Iterator selector,
-      const std::vector<cut_value_t>& cuts, const Logger& logger) {
-    const auto& etaBins = selector->etaBins;
-    if (etaBins.empty()) {
-      return cuts[0];  // shortcut if no etaBins
-    }
-    const auto eta = std::atanh(std::cos(trackState.predicted()[eBoundTheta]));
-    const auto abseta = std::abs(eta);
-    std::size_t bin = 0;
-    for (auto etaBin : etaBins) {
-      if (etaBin >= abseta) {
-        break;
-      }
-      bin++;
-    }
-    if (bin >= cuts.size()) {
-      bin = cuts.size() - 1;
-    }
-    ACTS_VERBOSE("Variable cut for eta=" << eta << ": " << cuts[bin]);
-    return cuts[bin];
-  }
+      const std::vector<cut_value_t>& cuts, const Logger& logger);
 
   double calculateChi2(
-      double* fullCalibrated, double* fullCalibratedCovariance,
-      TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
-                       false>::Parameters predicted,
-      TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
-                       false>::Covariance predictedCovariance,
-      TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
-                       false>::Projector projector,
-      unsigned int calibratedSize) const;
+      const double* fullCalibrated, const double* fullCalibratedCovariance,
+      const TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
+                             false>::Parameters predicted,
+      const TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
+                             false>::Covariance predictedCovariance,
+      const TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
+                             false>::Projector projector,
+      const unsigned int calibratedSize) const;
```

Comment:
```suggestion
      TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
                             false>::Parameters predicted,
      TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
                             false>::Covariance predictedCovariance,
      TrackStateTraits<MultiTrajectoryTraits::MeasurementSizeMax,
                             false>::Projector projector,
      unsigned int calibratedSize) const;
```

nitpick: `const` for value types have no effect in declaration